### PR TITLE
Homepage added to Cargo.toml

### PR DESCRIPTION
**Pull request by hiirrxnn**
Let me know if this works !

### CODE REVIEW

**Comment by Fokko** [comment]
@hiirrxnn I think we want to add the website to each of the `Cargo.toml` in each of the crates, for example here: https://github.com/apache/iceberg-rust/blob/b7031464dfd57dbfb9e872121377d3a6a6b401c5/crates/catalog/hms/Cargo.toml#L23-L27

**Comment by hiirrxnn** [comment]
Is it alright now ? I've done it in the way mentioned by the doc referenced here . Let me know if something has to be changed here .

**Comment by liurenjie1024** [comment]
Hi, @hiirrxnn There is another approach for this, please follow these steps:

1. You can add the home page info in `Cargo.toml` of workspace.
2. Then in `Cargo.toml` of each package, set the filed value  to `homepage = {workspace = true}`.

Here is a great example from cargo: https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table

This way we don't need to repeat same thing for each package.

**Comment by hiirrxnn** [comment]
I have done that ! Is that what you meant ? Let me know if there's something wrong !

**Comment by liurenjie1024** [comment]
> I have done that ! Is that what you meant ? Let me know if there's something wrong !

Yeah, exactly, thanks!

There are also several fields which are shared among all packages in a workspace:

1. version
2. edition
3. license
4. repository

Do you want to modify them together or we can do it in later prs, it's up to you.

**Comment by hiirrxnn** [comment]
I will start doing those other packages asap , but i think i should do that in a different pr as too much modified code in one pr might not be ideal . Please tell me how you think about this , otherwise i can do it in this pr as well !

**Comment by hiirrxnn** [comment]
Should i follow the exact same approach for all of them , in essence referring them from workspace Cargo.toml?

**Comment by liurenjie1024** [comment]
> Should i follow the exact same approach for all of them , in essence referring them from workspace Cargo.toml?

Yes, thanks!

**Comment by liurenjie1024** [comment]
> I will start doing those other packages asap , but i think i should do that in a different pr as too much modified code in one pr might not be ideal . Please tell me how you think about this , otherwise i can do it in this pr as well !

Cool, looking forward to your next pr!

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Comment by hiirrxnn** [comment]
Since this PR hadn't been merged yet , the new changes in new PR contains both original issue changes and new ones . Please check that out and let me know if something has to be changed !

**Review by Fokko — APPROVED** [review]
Nice!

File: Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,9 @@
 resolver = "2"
 members = ["crates/catalog/*", "crates/iceberg", "crates/test_utils"]
 
+[workspace.package]
+homepage = {url = "https://rust.iceberg.apache.org/"}
```
Comment (Fokko) [inline Cargo.toml:23]: I noticed that the CI is failing:
```
Caused by:
  TOML parse error at line 23, column 12
     |
  23 | homepage = {url = "https://rust.iceberg.apache.org/"}
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  invalid type: map, expected a string
```

```suggestion
homepage = "https://rust.iceberg.apache.org/"
```

**Review by Xuanwo — APPROVED** [review]
Thanks! It's better to run `taplo fmt` to format the toml files.

File: Cargo.toml (hunk 1 above)
Comment (hiirrxnn) [reply Cargo.toml:23]: I have fixed this in the new PR , please review those changes and let me know if something has to be improved

**Comment by Fokko** [comment]
I tried to resolve conflicts, but failed 😞

**Comment by liurenjie1024** [comment]
Seems duplicated with #160. How about we close this pr and focus on #160 only?

**Comment by liurenjie1024** [comment]
> Thanks! It's better to run `taplo fmt` to format the toml files.

Good suggestion!

**Pull request closed without merging**